ampersand-model.js

```diff
@@ -60,10 +88,18 @@ var Model = State.extend({
         wrapError(this, options);
 
         method = this.isNew() ? 'create' : (options.patch ? 'patch' : 'update');
-        if (method === 'patch') options.attrs = attrs;
+
+        if (method === 'patch') {
+            options.attrs = transformForPatch(new this.constructor(attrs).serialize(), attrs);
```

Comment:
Can you explain the reasoning behind instantiating a copy of ourselves to call serialize instead of just calling it on `this`?